test/unit/Cover/buyCover.js

```diff
@@ -802,4 +805,464 @@ describe('buyCover', function () {
+      {
+        value: expectedPremium,
+      },
+    );
+
+    const stakingPool1After = await tokenController.stakingPoolNXMBalances(0);
+    const stakingPool2After = await tokenController.stakingPoolNXMBalances(1);
+    const stakingPool3After = await tokenController.stakingPoolNXMBalances(2);
+    const stakingPool4After = await tokenController.stakingPoolNXMBalances(3);
+    const stakingPool5After = await tokenController.stakingPoolNXMBalances(4);
+
+    // validate that rewards increased
+    expect(stakingPool1After.rewards).to.be.gt(stakingPool1Before.rewards);
+    expect(stakingPool2After.rewards).to.be.gt(stakingPool2Before.rewards);
+    expect(stakingPool3After.rewards).to.be.gt(stakingPool3Before.rewards);
+    expect(stakingPool4After.rewards).to.be.gt(stakingPool4Before.rewards);
+    expect(stakingPool5After.rewards).to.be.gt(stakingPool5Before.rewards);
+
+    for (let i = 0; i < 5; i++) {
+      const segmentAllocation = await cover.coverSegmentAllocations(coverId, segmentId, i);
+      expect(segmentAllocation.poolId).to.be.equal(i);
+      expect(segmentAllocation.coverAmountInNXM).to.be.equal(coverAmountInAsset);
+    }
+  });
+
+  it('reverts if reentrant', async function () {
+    const { cover, memberRoles } = this;
+
+    const {
+      members: [coverBuyer, coverReceiver],
+    } = this.accounts;
+
+    const ReentrantExploiter = await ethers.getContractFactory('ReentrancyExploiter');
+    const reentrantExploiter = await ReentrantExploiter.deploy();
+    await memberRoles.setRole(reentrantExploiter.address, 2);
+
+    const { amount, productId, coverAsset, period, poolId, targetPriceRatio, priceDenominator } = buyCoverFixture;
+
+    const commissionRatio = 500; // 5%
+    const expectedBasePremium = amount
+      .mul(targetPriceRatio)
+      .div(priceDenominator)
+      .mul(period)
+      .div(3600 * 24 * 365);
+    const expectedCommission = expectedBasePremium.mul(commissionRatio).div(priceDenominator);
+    const expectedPremium = expectedBasePremium.add(expectedCommission);
+
+    const txData = await cover.connect(coverBuyer).populateTransaction.buyCover(
+      {
+        owner: coverReceiver.address,
+        productId,
+        coverAsset,
+        amount,
+        period,
+        maxPremiumInAsset: expectedPremium,
+        paymentAsset: coverAsset,
+        payWitNXM: false,
+        commissionRatio: parseEther('0'),
+        commissionDestination: AddressZero,
+        ipfsData: '',
+      },
+      [{ poolId, coverAmountInAsset: amount }],
+      {
+        value: expectedPremium,
+      },
+    );
+
+    await setEtherBalance(reentrantExploiter.address, expectedBasePremium.mul(2));
+    await reentrantExploiter.setFallbackParams([cover.address], [expectedBasePremium], [txData.data]);
+
+    // The test uses the payment to the commission destination to reentrant buyCover again.
+    // The nonReentrant protection will make revert that new call, making the payment to the commission address to fail.
```

Comment:
`The nonReentrant protection will make the new call revert` - correct rephrasing